assets/animations.js

```diff
@@ -36,7 +38,59 @@ function initializeScrollAnimationTrigger(rootEl = document, isDesignModeEvent =
   animationTriggerElements.forEach((element) => observer.observe(element));
 }
 
-window.addEventListener('DOMContentLoaded', () => initializeScrollAnimationTrigger());
+// Zoom in animation logic
+function initializeScrollZoomAnimationTrigger() {
+  const animationTriggerElements = Array.from(document.getElementsByClassName(SCROLL_ZOOM_IN_TRIGGER_CLASSNAME));
+  const scaleAmount = 0.2 / 100;
+  if (animationTriggerElements.length === 0) return;
+
+  let scrollPositionY = 0;
+
+  animationTriggerElements.forEach((element) => {
+    let elementIsVisible = false;
+    const observer = new IntersectionObserver((elements) => {
+      elements.forEach((entry) => {
+        elementIsVisible = entry.isIntersecting;
+      });
+    });
+    observer.observe(element);
+
+    element.style.setProperty('--zoom-in-ratio', 1 + scaleAmount * percentageSeen(element));
+
+    window.addEventListener('scroll', () => {
```

Comment:
Ideally, we'd want to throttle all scroll event listeners.

Reply:
Let me know if you think `100` is enough. It doesn't seem to impact too poorly the animation this way 👍